docs/adr/0004-bundling-experiments.md

```diff
@@ -0,0 +1,74 @@
+    * We can retrieve the experiment data on the first run, experiment data however will not be available until after a short delay (network I/O + some disk I/O)
+
+## Decision Outcome
+
+The proposed option: **(B) Bundle Experiment data with app on release** was **rejected**
+
+The option was rejected mainly due to difficulty in disabling experiments and pausing enrollments. This can create a negative user experience as it prevents us from disabling any problematic experiments. Additionally, it ties experiment creation with application release cycles.
+
+Moving forward, we are discussing **(C) Retrieve Experiment data on first run, and deal with delay** to address the first product experimentation requirement, which involves the ability to run experiments on a user's first run, but not necessarily early (The first experiments don't happen to involve onboarding)
+
+### Positive Consequences
+
+* Since the experiments are bundled with the app, they will always be available on first run, without having to wait on network I/O. This allows us to create experiments early on the user's first run (for example, onboarding experiments)
+
+### Negative Consequences
+
+* Inability to disable experiments without issuing a dot-release. The hinders our ability to quickly rollback any experiments that are causing problems. As a consequence, this will slow us down in that we would not be able to launch experiments and iterate quickly if it goes wrong
+* Inability to pause enrollments remotely. We currently have a mechanism to pause enrollments directly on experimenter, however, since the experiment data from remote-settings is not applied until the second run, users would still get enrolled in experiments (whose enrollment has been paused) on their first run.
+* Experiment creation will be tied to release cycles, any experiments that would like to target users on their first run would need to be created before a release is made. This removes flexibility in experiment creation and creates limitation on the timeline experiments can be created to be tied with release cycles (or would force us to create dot-releases for experiments)
+
+## Pros and Cons of the Options
+
+### Do nothing
+
+* Good, because it keeps the flexibility in experiment creation
+* Good, because disabling experiments can still done remotely for all experiments
+* Bad, because it doesn't address the main problem of exposing experiments to user on their first run
+
+### Bundle Experiment data with app on release
+* Good, because it allows us to run experiments early on a user's first run
+* Good, because it prevents us from having to wait for experiments, especially if a user has a slow network connection
+* Bad, because it ties experiment creation with release cycles
+* Bad, because it prevents us from disabling problematic first-run experiments without a dot release
+* Bad, because it prevents us from pausing enrollment on first-run experiments without a dot release
```

Comment:
The logic for the `problematic first-run experiments` is that since we bundle them, they'll be loaded on the first run (and since they're first run experiments users might get enrolled) - without bundling, new users wouldn't experience the bad experiments at all if they're disabled on experimenter.

I'll make sure to specify that pausing is still possible! (and that this only limits pausing remotely)